infra/modules/providers/azure/app-gateway/variables.tf

```diff
@@ -0,0 +1,107 @@
+variable "resource_group_name" {
+  description = "Resource group name that the app gateway will be created in."
+  type        = "string"
+}
+
+variable "virtual_network_name" {
+  description = "Resource group name that the app gateway will be created in."
+  type        = "string"
+}
+
+variable "public_ip_name" {
+  description = "Resource group name that the app gateway will be created in."
```

Comment:
There will be scenarios where a customer will want to keep their app gateway private. Can we make this an optional variable?

Reply:
I tried this scenario but that wouldn't work since we have a data block for Public IP
- We might have to either remove the data block and provide a variable or 
- keep it a must for customer to provide public IP.